modules/nixvim/nixvim.nix

```diff
@@ -11,48 +11,74 @@
       main = lib.mkEnableOption "background transparency for the main NeoVim window";
       sign_column = lib.mkEnableOption "background transparency for the NeoVim sign column";
     };
+    config = lib.mkOption {
+      type = with lib.types; attrsOf anything;
+      readOnly = true;
+
+      description = ''
+        The stylix configuration, generated for nixvim.
+
+        If nixvim is installed via nixos, darwin, or home-manager then this will be **automatically**
+        assigned to `programs.nixvim`. If you're using a "standalone" build of nixvim, then that's
+        not possible. Instead, you should pass this config to the `nixvimExtend` function.
+
+        For example:
+
+        ```nix
+          { config, ... }: {
+            environment.systemPackages = [
+              (standalone-nixvim-derivation.nixvimExtend config.stylix.targets.nixvim.config)
+            ];
+          }
+        ```
+
+        See nixvim's docs on [extending a standalone configuration](https://nix-community.github.io/nixvim/modules/standalone.html#extending-an-existing-configuration).
+      '';
+    };
   };
 
-  config = lib.mkIf ((config.programs ? nixvim) && config.stylix.targets.nixvim.enable) (
-    lib.optionalAttrs (builtins.hasAttr "nixvim" options.programs) {
-      programs.nixvim = {
-        colorschemes.base16 = {
-          colorscheme = let
-            colors = config.lib.stylix.colors.withHashtag;
-          in {
-            base00 = colors.base00;
-            base01 = colors.base01;
-            base02 = colors.base02;
-            base03 = colors.base03;
-            base04 = colors.base04;
-            base05 = colors.base05;
-            base06 = colors.base06;
-            base07 = colors.base07;
-            base08 = colors.base08;
-            base09 = colors.base09;
-            base0A = colors.base0A;
-            base0B = colors.base0B;
-            base0C = colors.base0C;
-            base0D = colors.base0D;
-            base0E = colors.base0E;
-            base0F = colors.base0F;
-          };
-
-          enable = true;
-        };
+  config = {
+    programs = lib.optionalAttrs (options.programs ? nixvim) (lib.mkIf config.stylix.targets.nixvim.enable {
+      nixvim = config.stylix.targets.nixvim.config;
```

Comment:
Apologies, I'm not sure I follow.

If you mean the read-only option should be named `programs.nixvim`, then that'll conflict with nixvim's option ("option declared twice" error).

If you mean `programs.nixvim` should be nested _within_ the final config's value, that wouldn't work because nixvim's **standalone** config modules _does not_ have the `programs.nixvim` prefix in its option paths.

If you mean we should do something like this, then sure - personal preference:

```nix

{
  options = /* ... */;
  config.programs.nixvim = lib.optionalAttrs (/* ... */) (lib.mkIf {/* ... */});
  config.stylix.targets.nixvim.config = {/* ... */};
}
```

If you mean `stylix.targets.nixvim.config` should change to something like `stylix.programs.nixvim` or `stylix.configs.nixvim` or `stylix.out.nixvim` (etc) then I'm open to that.

If you mean something else, would you mind clarifying?

Reply:
If by online, you mean we should remove the read-only option; adding that option is the purpose of this PR.

By extracting the attrset intended for nixvim, we allow (advanced) nixvim users _not_ using nixvim's home-manager/NixOS modules to still apply stylix's config to a _standalone_ nixvim (using the derivation's `extendNixvim` function)

Reply:
> Apologies, I'm not sure I follow.

Sorry for the imprecise explanation.

> If you mean the read-only option should be named `programs.nixvim`, then that'll conflict with nixvim's option ("option declared twice" error).
>
> If you mean `programs.nixvim` should be nested _within_ the final config's value, that wouldn't work because nixvim's **standalone** config modules _does not_ have the `programs.nixvim` prefix in its option paths.
>
> If you mean we should do something like this, then sure - personal preference:
>
> ```nix
> {
>   options = /* ... */;
>   config.programs.nixvim = lib.optionalAttrs (/* ... */) (lib.mkIf {/* ... */});
>   config.stylix.targets.nixvim.config = {/* ... */};
> }
> ```

Thanks for the detailed analysis! Based on that, your initial proposal seems to be the right one.

> If you mean `stylix.targets.nixvim.config` should change to something like `stylix.programs.nixvim` or `stylix.configs.nixvim` or `stylix.out.nixvim` (etc) then I'm open to that.

What about interfacing the `config` under the `config.lib.stylix` namespace, similar to:

https://github.com/danth/stylix/blob/1d3826ceed91ae67562f28ee2e135813a11e47a6/stylix/palette.nix#L149-L150

In that case, `config.lib.stylix.targets.nixvim.config` sounds good to me. @danth, what do you think?

Reply:
Making the read-only option a `stylix.lib` suboption would probably be neater. Although, having `stylix.configs` (a sibling of `stylix.lib`) would be even better (IMO): e.g. `stylix.configs.nixvim`. This pattern could then also be applied to other targets as `stylix.configs.*`.

This makes some sense to me, as it's kinda analogous to the final `config` attrset, which is passed as a module arg alongside `lib`, `pkgs`, `options`, etc.

I think if we go down this route, then it makes more sense to have it documented separately:\
The advantage of having the option under `stylix.targets.nixvim` was discoverability - anyone looking at the nixvim `enable` option will stumble on the `config` option - so if they're grouped elsewhere we'd have to improve discoverability through docs (at a later date).